### PR TITLE
remove RunPlan class that wraps Task

### PR DESCRIPTION
**Pull request by stan-dot**
quick renaming 36 refs and delete the class

### CODE REVIEW

File: src/blueapi/worker/task.py [hunk 1]
```diff
@@ -32,16 +32,6 @@ def do_task(self, ctx: BlueskyContext) -> None:
         ctx.run_engine(wrapped_plan_generator)
 
 
-# Here for backward compatibility pending
-# https://github.com/DiamondLightSource/blueapi/issues/253
-class RunPlan(Task):
```
Comment (DiamondJoseph) [inline src/blueapi/worker/task.py:37]: Is this still required for backwards compatibility? Is it referenced anywhere?

File: docs/user/reference/openapi.yaml
Comment (DiamondJoseph) [inline docs/user/reference/openapi.yaml:1]: How did you generate this file? It'd be good to remove all the unrelated formatting only changes.

Comment (stan-dot) [reply docs/user/reference/openapi.yaml:1]: I think I just ran ruff / pre-commit automatically

File: src/blueapi/worker/task.py (hunk 1 above)
Comment (stan-dot) [reply src/blueapi/worker/task.py:37]: to be honest RunEngineWorker shares a name, but might be renamed into TaskWorker

Comment (stan-dot) [reply src/blueapi/worker/task.py:37]: renamed

File: docs/user/reference/openapi.yaml
Comment (stan-dot) [reply docs/user/reference/openapi.yaml:1]: now I auto-generated a new one using 
`python -m blueapi.cli schema -u`.
formatting indeed has changed
